cluster-autoscaler/simulator/cluster_snapshot_test.go

```diff
@@ -224,6 +224,132 @@ func TestReAddNode(t *testing.T) {
 	}
 }
 
+type snapshotState struct {
+	nodes []*apiv1.Node
+	pods  []*apiv1.Pod
+}
+
+func compareStates(t *testing.T, a, b snapshotState) {
+	assert.ElementsMatch(t, a.nodes, b.nodes)
+	assert.ElementsMatch(t, a.pods, b.pods)
+}
+
+func getSnapshotState(t *testing.T, snapshot ClusterSnapshot) snapshotState {
+	nodes, err := snapshot.NodeInfos().List()
+	assert.NoError(t, err)
+	pods, err := snapshot.Pods().List(labels.Everything())
+	assert.NoError(t, err)
+	return snapshotState{extractNodes(nodes), pods}
+}
+
+func startSnapshot(t *testing.T, snapshotFactory func() ClusterSnapshot, state snapshotState) ClusterSnapshot {
+	snapshot := snapshotFactory()
+	err := snapshot.AddNodes(state.nodes)
+	assert.NoError(t, err)
+	for _, pod := range state.pods {
+		err := snapshot.AddPod(pod, pod.Spec.NodeName)
+		assert.NoError(t, err)
+	}
+	return snapshot
+}
+
+func TestForking(t *testing.T) {
```

Comment:
+1 I think this is a nice test setup that allows to easily cover a lot of cases.